php/example_code/apigateway/Get_BasePathmapping.php

```diff
@@ -1,65 +1,135 @@
+    and activate a new Amazon Web Services account" on the AWS Premium Support
+    website.
+  - This code uses default AWS access credentials. For more information, see
+    "Credentials for the AWS SDK for PHP" in the AWS SDK for PHP Developer 
+    Guide.
+  - You must have a custom domain name in API Gateway. For more information, 
+    see "Custom Domain Names" in the Amazon API Gateway Developer Guide.
+
+Running the code:
+  To run this code, use PHPUnit along with the phpunit.xml file in this folder.
+  For example:
+
+  ./vendor/bin/phpunit --testsuite apigateway-getbasepathmapping
+
+Additional information:
+  - As an AWS best practice, grant this code least privilege, or only the 
+    permissions required to perform a task. For more information, see 
+    "Grant Least Privilege" in the AWS Identity and Access Management 
+    User Guide.
+  - This code has not been tested in all AWS Regions. Some AWS services are 
+    available only in specific Regions. For more information, see the 
+    "AWS Regional Table" on the AWS website.
+  - Running this code outside of the included PHPUnit test might result in 
+    charges to your AWS account.
+
+//////////////////////////////////////////////////////////////////////////// */
+
 // snippet-start:[apigateway.php.get_base_path_mapping.complete]
 // snippet-start:[apigateway.php.get_base_path_mapping.import]
-
 require 'vendor/autoload.php';
 
-use Aws\ApiGateway\ApiGatewayClient;   
+use Aws\ApiGateway\ApiGatewayClient;
 use Aws\Exception\AwsException;
 // snippet-end:[apigateway.php.get_base_path_mapping.import]
 
-//Create a ApiGatewayClient 
 //snippet-start:[apigateway.php.get_base_path_mapping.main]
-$client = new new Aws\ApiGateway\ApiGatewayClient([
-    'profile' => 'default',
-    'version' => '2015-07-09',
-    'region' => 'us-east-2'
-]);
-
-$basePath = '(none)';
-$domainName = 'example.com';
-
-try {
-    $result = $client->getBasePathMapping([
-        'basePath' => $basePath,
-        'domainName' => $domainName,
-    ]);
-    var_dump($result);
-} catch (AwsException $e) {
-    // output error message if fails
-    echo $e->getMessage();
-    echo "\n";
-}
- 
- 
+class GetBasePathMappingExample
+{
+    private $apiGatewayClient = null;
+
+    public function __construct($apiGatewayClient)
+    {
+        $this->apiGatewayClient = $apiGatewayClient;
+    }
+    
+    /* ////////////////////////////////////////////////////////////////////////
+
+    Purpose: Gets the base path mapping for a custom domain name in API Gateway.
+
+    Inputs:
+      - $basePath: The base path name that callers must provide as part of the 
+        URL after the domain name.
+      - $domainName: The custom domain name for the base path mapping.
+
+    Returns: true if the API call succeeds; otherwise, false.
+
+    //////////////////////////////////////////////////////////////////////// */
+    public function getBasePathMapping($basePath, $domainName)

```

Comment:
I'm not sure what value add to the customer is here of wrapping the SDK operation in a wrapper method that just invokes the SDK operation, and then returns less specific information (i.e. true/false instead of the actual mapping data). Note that returning true/false does not actually return what the `Purpose` claims.

The original code seemed more succinct and, at least to me, more helpful as a reference. If the goal is to add meatier examples, perhaps something more realistic would be less confusing?